client/components/ProductSingle.js

```diff
@@ -1,37 +1,37 @@
 /* eslint-disable react/destructuring-assignment */
+/* eslint-disable no-shadow */
 import React from 'react';
-import axios from 'axios';
-import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { fetchProductReviews, fetchProduct } from '../store';
 
-class ProductSingle extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      product: {},
-      reviews: [],
-    };
-  }
-
-  componentDidMount() {
-    // TODO: what's a more efficient way of doing this (that linter likes)?
-    const { match } = this.props;
-    const { params } = match;
-    const { productId } = params;
+const mapStateToProps = (state, props) => {
+  const { products } = state.product;
+  const { reviews } = state.review;
+  return {
+    products,
+    reviews,
+  };
+};
 
-    axios
-      .get(`api/products/${productId}`)
-      .then(res => res.data)
-      .then(product => this.setState({ product }));
+const mapDispatchToProps = dispatch => {
+  return {
+    fetchProductReviews: id => dispatch(fetchProductReviews(id)),
+    fetchProduct: id => dispatch(fetchProduct(id)),
+  };
+};
 
-    axios
-      .get(`api/reviews/${productId}`)
-      .then(res => res.data)
-      .then(reviews => this.setState({ reviews }));
-  }
+class ProductSingle extends React.Component {
+  componentDidMount = () => {
+    const { productId } = this.props.match.params;
+    const { fetchProduct, fetchProductReviews } = this.props;
+    fetchProductReviews(productId);
+    fetchProduct(productId);
+  };
 
   render() {
-    const { product, reviews } = this.state;
-
+    const { products, reviews } = this.props;
```

Comment:
See my comments in store about creating a product reducer, along with action type and action creator.
```suggestion
    const { product, reviews } = this.props;
```